### PR TITLE
Copy api/OWNERS to docs/api-reference

### PR DESCRIPTION
**Pull request by tallclair**
So that API changes can be approved by API approvers. Also, create api-{approvers,reviewers} aliases.

Suggested by @cblecker in https://github.com/kubernetes/kubernetes/pull/53911#issuecomment-336612394

### CODE REVIEW

**Comment by cblecker** [comment]
we don't have a way of referencing another directory's OWNERS file, but we do have the [`OWNERS_ALIASES`](https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES) file that allows you to create a group (say "api-reviewers" and "api-approvers") and then reference that group in multiple places. (e.g. how sig-scheduling does it https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/OWNERS )

**Comment by liggitt** [comment]
`verify-api-reference-docs.sh` needs updating to ignore the OWNERS file when diffing

File: staging/src/k8s.io/api/core/OWNERS [hunk 1]
```diff
@@ -1,42 +1,4 @@
 approvers:
-- erictune
-- lavalamp
-- smarterclayton
-- thockin
+- api-approvers
```
Comment (tallclair) [inline staging/src/k8s.io/api/core/OWNERS]: This adds @liggitt here.

**Comment by cblecker** [comment]
@tallclair as @liggitt noted, the following line needs `--exclude=OWNERS` added to it:
https://github.com/kubernetes/kubernetes/blob/5adfb24f8f25a0d57eb9a7b158db46f9f46f0d80/hack/verify-api-reference-docs.sh#L39

**Review by tallclair — COMMENTED** [review]
Fixed kubernetes/hack/verify-api-reference-docs.sh

File: staging/src/k8s.io/api/core/OWNERS (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/core/OWNERS]: covered by `staging/src/k8s.io/api/OWNERS` already

**Comment by cblecker** [comment]
/approve no-issue
For `hack/verify-api-reference-docs.sh` change

File: staging/src/k8s.io/api/core/OWNERS (hunk 1 above)
Comment (tallclair) [reply staging/src/k8s.io/api/core/OWNERS]: Removed.

File: staging/src/k8s.io/api/OWNERS [hunk 2]
```diff
@@ -1,51 +1,11 @@
 approvers:
-- erictune
-- lavalamp
-- smarterclayton
-- thockin
-- liggitt
-# - bgrant0607 # manual escalations only
+- api-approvers
```
Comment (liggitt) [inline staging/src/k8s.io/api/OWNERS]: for staging components pushed out to other repos, the OWNERS_ALIASES file in the root will be lost. not sure we want to reference that here

Comment (tallclair) [reply staging/src/k8s.io/api/OWNERS]: Does that matter, since no one should be making changes in those repos? (Should we be copying OWNERS files at all?)

Comment (liggitt) [reply staging/src/k8s.io/api/OWNERS]: The goal is to shape them so they can be standalone repos, even if dev and CI haven't caught up to allow that yet

**Comment by cblecker** [comment]
```
W1019 05:18:23.362] docker: Error response from daemon: Get https://gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
```
Trying again as gcr flakes aren't common.
/test pull-kubernetes-unit

File: staging/src/k8s.io/api/OWNERS (hunk 2 above)
Comment (tallclair) [reply staging/src/k8s.io/api/OWNERS]: Reverted.

**Pull request merged**